api/tests/test_audit_log.py

```diff
@@ -0,0 +1,176 @@
+    delivery_log = DeliveryLog.objects.latest("pk")
+
+    # Assert that an AuditLogEntry was created for the creation
+    qs = AuditLogEntry.objects.filter(
+        message__audit_event__target__path=url,
+        message__audit_event__operation=Operation.CREATE.value,
+    )
+    assert qs.count() == 1
+    audit_log_entry = qs.first()
+
+    # Verify the audit log entry details
+    assert audit_log_entry.message["audit_event"]["target"]["object_ids"] == [
+        str(delivery_log.pk)
+    ]
+    assert audit_log_entry.message["audit_event"]["target"]["path"] == url
+    assert audit_log_entry.message["audit_event"]["operation"] == Operation.CREATE.value
+
+
+def test_updating_delivery_log_writes_update_log(admin_user):
+    """
+    Test that updating a DeliveryLog object writes an update log.
+    """
+    client = Client()
+    client.force_login(admin_user)
+
+    delivery_log = DeliveryLogFactory()
+
+    # Get the URL for updating the DeliveryLog object
+    url = reverse("admin:api_deliverylog_change", args=[delivery_log.pk])
+
+    # Send a POST request with updated data
+    updated_data = {
+        "user": admin_user.pk,
+        "report": "null",
+    }  # Example updated data
+    response = client.post(url, updated_data)
+    assert response.status_code == 302  # Redirect after successful update
+
+    # Assert that an AuditLogEntry was created for the update
+    qs = AuditLogEntry.objects.filter(
+        message__audit_event__target__path=url,
+        message__audit_event__operation=Operation.UPDATE.value,
+    )
+    assert qs.count() == 1
+    audit_log_entry = qs.first()
+
+    # Verify the audit log entry details
+    assert audit_log_entry.message["audit_event"]["target"]["object_ids"] == [
+        str(delivery_log.pk)
+    ]
+    assert audit_log_entry.message["audit_event"]["target"]["path"] == url
+    assert audit_log_entry.message["audit_event"]["operation"] == Operation.UPDATE.value
+
+
+def test_deleting_delivery_log_writes_delete_log(admin_user):
+    """
+    Test that deleting a DeliveryLog object writes a delete log.
+    """
+    client = Client()
+    client.force_login(admin_user)
+
+    delivery_log = DeliveryLogFactory()
+
+    # Get the URL for deleting the DeliveryLog object
+    url = reverse("admin:api_deliverylog_delete", args=[delivery_log.pk])
+
+    # Send a POST request to delete the object
+    response = client.post(url, {"post": "yes"})
+    assert response.status_code == 302  # Redirect after successful deletion
+
+    # Assert that an AuditLogEntry was created for the deletion
+    qs = AuditLogEntry.objects.filter(
+        message__audit_event__target__path=url,
+        message__audit_event__operation=Operation.DELETE.value,
+    )
+    assert qs.count() == 1
+    audit_log_entry = qs.first()
+
+    # Verify the audit log entry details
+    assert audit_log_entry.message["audit_event"]["target"]["object_ids"] == [
+        str(delivery_log.pk)
+    ]
```

Comment:
Same comment here as in https://github.com/City-of-Helsinki/notification-service-api/pull/69/files#r1843403523
```suggestion
    assert sorted(
        audit_log_entry.message["audit_event"]["target"]["object_ids"]
    ) == sorted(str(delivery_log.pk))
```